src/FixedPointNumbers.jl

```diff
@@ -260,27 +261,49 @@ function length(r::StepRange{<:FixedPoint})
     return div((stop - start) + step, step)
 end
 
+hasalias(::Type) = false
+hasalias(::Type{X}) where {T<:NotBiggerThanInt64, f, X<:FixedPoint{T,f}} = nbitsfrac(X) isa Int
+
 # Printing. These are used to generate type-symbols, so we need them
-# before we include any files.
-function showtype(io::IO, ::Type{X}) where {X <: FixedPoint}
-    print(io, typechar(X))
-    f = nbitsfrac(X)
-    m = bitwidth(X)-f-signbits(X)
-    print(io, m, 'f', f)
+# before we include "src/fixed.jl" / "src/normed.jl".
+@inline function showtype(io::IO, ::Type{X}) where {X <: FixedPoint}
+    if hasalias(X)
+        f = nbitsfrac(X)
+        m = nbitsint(X)
+        write(io, typechar(X))
+        m > 9 && write(io, Char(m ÷ 10 + 0x30))
+        write(io, Char(m % 10 + 0x30), 'f')
+        f > 9 && write(io, Char(f ÷ 10 + 0x30))
+        write(io, Char(f % 10 + 0x30))
+    else
+        print(io, X)
+    end
     io
 end
+
 function show(io::IO, x::FixedPoint{T,f}) where {T,f}
+    compact = get(io, :compact, false)
```

Comment:
```suggestion
    compact = get(io, :compact, false)::Bool
```
There is no way for inference to know this, so it makes sense to help it.

It's not necessary for the `:typeinfo` access below because the only way you use it is with `===` and that's a builtin so there's only one "method."

Reply:
BTW, in `Base`, the default value for `:typeinfo` seems to be set `Any`, not `nothing`. So, I'll change it to `Any`.